www/source/blog/2017-08-09-Habitat-plans-for-net-full-iis-apps.md

```diff
@@ -0,0 +1,301 @@
+    ."$env:SystemRoot\System32\inetsrv\appcmd.exe" add app /site.name:"{{cfg.site_name}}" /path:"/{{cfg.app_name}}" /PhysicalPath:"$((Resolve-Path '{{pkg.svc_var_path}}').Path)" /applicationPool:"{{cfg.app_pool}}"
+}
+
+."$env:SystemRoot\System32\inetsrv\appcmd.exe" start apppool "{{cfg.app_pool}}"
+."$env:SystemRoot\System32\inetsrv\appcmd.exe" start site "{{cfg.site_name}}"
+```
+
+This uses values in our configuration and creates the app pools, site and application folders needed to run our site.
+
+### DSC
+
+If you are familiar with [DSC](https://docs.microsoft.com/en-us/powershell/dsc/overview) and particularly using the resources in `xWebAdministration` then using Habitat to configure IIS should come natrually. We will create a `config` folder in our `habitat` directory and a `website.ps1` inside of that. Here you might have the following configuration:
+
+```powershell
+Configuration NewWebsite
+{
+    Import-DscResource -Module xWebAdministration
+    Node 'localhost' {
+
+        xWebAppPool {{cfg.app_pool}}
+        {
+            Name   = "{{cfg.app_pool}}"
+            Ensure = "Present"
+            State  = "Started"
+        }
+        
+        xWebsite {{cfg.site_name}}
+        {
+            Ensure          = "Present"
+            Name            = "{{cfg.site_name}}"
+            State           = "Started"
+            PhysicalPath    = Resolve-Path "{{pkg.svc_path}}"
+            ApplicationPool = "{{cfg.app_pool}}"
+            BindingInfo = @(
+                MSFT_xWebBindingInformation
+                {
+                    Protocol = "http"
+                    Port = {{cfg.port}}
+                }
+            )
+        }
+
+        xWebApplication {{cfg.app_name}}
+        {
+            Name = "{{cfg.app_name}}"
+            Website = "{{cfg.site_name}}"
+            WebAppPool =  "{{cfg.app_pool}}"
+            PhysicalPath = Resolve-Path "{{pkg.svc_var_path}}"
+            Ensure = "Present"
+        }
+    }
+}
+```
+
+This uses our Habitat templating capabilities to ensure that IIS points to where our application is stored in the Habitat `svc` folder and uses the configured port.
+
+Our `run` hook will apply this configuration whenever we start our ASP.NET application and make sure everything is configured according to our DSC or `appcmd.exe` calls.
+
+## Running our application in the Habitat supervisor
+
+In full framewoerk apps, IIS is often in charge of the process lifecycle of our app unlike Node or .NET Core where we would simply invoke `node.exe` or `dotnet.exe` to run our app in a light weight web server. So instead of handing off our run hook to call into a runtime we just need to make sure IIS is configured and the site and app pool is in the "running" state. We already covered IIS configuration above. However if the run hook ends, the supervisor assumes our service has terminated. So instead, we will have powershell loop and check that our application endpoint is responsive:
```

Comment:
"full framework" => "full-framework" ?

"...lifecycle of our app unlike Node..." => "...lifecycle of our app, unlike Node..."

"light weight" => "lightweight"

"run hook" => "`run` hook"

"call into a runtime we just need" => "call into a runtime, we just need"

"and the site and app pool is in the "running" state." => "and the site and app pool are both in the "running" state."

"However if the run hook ends, " => "However, if the `run` hook terminates, "

"So instead, we will " => "As a result, will will"

"powershell" => "PowerShell"